acme/setup.py

```diff
@@ -26,6 +27,12 @@
     'six>=1.9.0',  # needed for python_2_unicode_compatible
 ]
 
+setuptools_known_environment_markers = (StrictVersion(setuptools_version) >= StrictVersion('36.2'))
+if setuptools_known_environment_markers:
+    install_requires.append('mock ; python_version < "3.3"')
+elif sys.version_info < (3,3):
```

Comment:
nit: I think we should probably error out if we're building the wheel and don't have `setuptools>=36.2` like we do [here](https://github.com/certbot/certbot/blob/e9895d2ec69395cfcf1cfff35e7668b57e1da00f/certbot/setup.py#L59-L74).

The reason I think this is if we build the wheel on Python 3.3+ using an old version of `setuptools`, the wheel we push to PyPI will never specify a dependency on `mock`. If people install this wheel on Python 2.7 and don't already happen to have `mock`, our tests will break.

I think this scenario is unlikely to come up for a number of reasons, but I personally think it's worth the two lines to make sure our dependencies are always specified properly.

Reply:
Great, I love it. I originally considered that and decided to do it this way because I understood the requirements as having it be more important to support a wider range of machines for people building the wheels (who are more likely to be the ones running the tests), but I absolutely am in favor of supporting a more narrow range of build environment setups.

Reply:
and to be clear, for systems with `sys.version_info < (3,3)` and old setuptools, would you rather just add `mock` or error out as well?

Reply:
I personally think we should add `mock` and not error out in the case of `sys.version_info < (3,3)` and old setuptools when we're not building a wheel.

My understanding is erroring out when building the wheel doesn't affect our packagers because they don't use our wheels or build their own, but erroring out in other cases would break things like our EPEL 7 packages which still use Python 2 and an old version of setuptools. I personally think it's worth the two lines to keep things working there for now.